tools/src/wyvern/tools/typedAST/abs/Declaration.java

```diff
@@ -93,4 +93,6 @@ public Declaration getNextDecl() {
 	public abstract DeclType genILType(GenContext ctx);
 
 	public abstract wyvern.target.corewyvernIL.decl.Declaration generateDecl(GenContext ctx, GenContext thisContext);
+	
+	public abstract wyvern.target.corewyvernIL.decl.Declaration topLevelGen(GenContext ctx);
```

Comment:
Can you add javadoc distinguishing what topLevelGen does from what generateDecl does?